### PR TITLE
Handle dev-session errors for each extension

### PR DESCRIPTION
**Pull request by isaacroldan**
### WHY are these changes introduced?

Fixes #0000 

### Measuring impact

How do we know this change was effective? Please choose one:

- [ ] n/a - this doesn't need measurement, e.g. a linting rule or a bug-fix
- [ ] Existing analytics will cater for this addition
- [ ] PR includes analytics changes to measure impact

### Checklist

- [ ] I've considered possible cross-platform impacts (Mac, Linux, Windows)
- [ ] I've considered possible [documentation](https://shopify.dev) changes

### CODE REVIEW

File: packages/app/src/cli/services/dev/processes/dev-session.ts [hunk 1]
```diff
@@ -145,13 +142,16 @@ async function handleDevSessionResult(
...
-  } else {
-    await printError(`❌ Error`, processOptions.stderr)
-    await printError(`└  ${result.error}`, processOptions.stderr)
+    outputDebug('❌ Session update aborted (new change detected or error in Session Update)', processOptions.stdout)
+  } else if (result.status === 'remote-error' || result.status === 'unknown-error') {
+    await processUserErrors(result.error, processOptions, processOptions.stdout)
   }
+
+  // If we failed to create a session, exit the process
+  if (!isDevSessionReady) process.exit(1)
```
Comment (shauns) [inline packages/app/src/cli/services/dev/processes/dev-session.ts:154]: Should this not be an AbortError or such?

File: packages/app/src/cli/services/dev/processes/dev-session.ts [hunk 2]
```diff
@@ -233,17 +254,20 @@ async function printWarning(message: string, stdout: Writable) {
...
 }
 
 async function printSuccess(message: string, stdout: Writable) {
   await printLogMessage(outputContent`${outputToken.green(message)}`.value, stdout)
 }
 
 // Helper function to print to terminal using output context with stripAnsi disabled.
-async function printLogMessage(message: string, stdout: Writable) {
-  await useConcurrentOutputContext({outputPrefix: 'dev-session', stripAnsi: false}, () => {
+async function printLogMessage(message: string, stdout: Writable, prefix?: string) {
```
Comment (shauns) [inline packages/app/src/cli/services/dev/processes/dev-session.ts:269]: Possible to add a test to cover the overall change (an error tied to an extension should output differently)?

**Review by shauns — CHANGES_REQUESTED** [review]

File: packages/app/src/cli/services/dev/processes/dev-session.ts (hunk 1 above)
Comment (isaacroldan) [reply packages/app/src/cli/services/dev/processes/dev-session.ts:154]: Yeah, that makes more sense, changed

**Review by shauns — APPROVED** [review]

**Comment by isaacroldan** [comment]
### Merge activity

* **Nov 22, 6:01 PM EST**: A user started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/Shopify/cli/4864).

**Pull request merged**